Makefile.frag

```diff
@@ -1,34 +1,45 @@
 
-INSTALL_DIR=$(shell pwd)/build/install
+BUILD_DIR=$(shell pwd)/build
+DEPS_DIR=$(BUILD_DIR)/deps
+INT_DIR=$(BUILD_DIR)/install
+INSTALL_DIR=$(shell pwd)
 
 # download and build libcrypto.a for the CRT with -fPIC
-build/deps/openssl:
+$(DEPS_DIR)/openssl:
 	@echo Fetching libcrypto source
-	git clone --depth=1 --single-branch --branch OpenSSL_1_1_1-stable https://github.com/openssl/openssl.git build/deps/openssl
+	git clone --depth=1 --single-branch --branch OpenSSL_1_1_1-stable https://github.com/openssl/openssl.git $(DEPS_DIR)/openssl
 
-$(INSTALL_DIR)/lib/libcrypto.a: build/deps/openssl
+$(INT_DIR)/lib/libcrypto.a: $(DEPS_DIR)/openssl
 	@echo Building libcrypto
-	cd build/deps/openssl && \
+	cd $(DEPS_DIR)/openssl && \
 		./config -fPIC no-md2 no-rc5 no-rfc3779 no-sctp no-ssl-trace no-zlib no-hw no-mdc2 no-seed no-idea \
 			no-camellia no-bf no-dsa no-ssl3 no-capieng \
 			-DSSL_FORBID_ENULL -DOPENSSL_NO_DTLS1 -DOPENSSL_NO_HEARTBEATS \
-			--prefix=$(INSTALL_DIR) --openssldir=$(INSTALL_DIR) && \
+			--prefix=$(INT_DIR) --openssldir=$(INT_DIR) && \
 		make -j && \
 		make install_sw
 
 
-CMAKE_CONFIGURE = cmake -DCMAKE_INSTALL_PREFIX=$(INSTALL_DIR) -DCMAKE_PREFIX_PATH=$(INSTALL_DIR) -DBUILD_TESTING=OFF
+CMAKE_CONFIGURE = cmake -DCMAKE_INSTALL_PREFIX=$(INT_DIR) -DCMAKE_PREFIX_PATH=$(INT_DIR) -DBUILD_TESTING=OFF
```

Comment:
I noticed this fails in Amazon Linux 2 since the requirement in those files in cmake 3.10+ and in Amazon Linux 2, the default cmake is 2.10, the 3+ cmake is installed under the `cmake3` command name. We may need to add something here to identify the right cmake to use.

Reply:
Fixed, did a check for cmake3

Reply:
GREAT!